File: c/fib/fib.c
// フィボナッチ数
#include <stdio.h>
#include <stdlib.h>

int fib(int n)
    {
        if (n < 2) {
            return n;
        } else {
            return fib(n-1) + fib(n-2);
        }
    }

int main(int argc, char *argv[])
    {
        int n;
        n = atoi(argv[1]);
        printf("%d", fib(n));
        return 0;
    }

/*
  Example
      compile:
              gcc -O3 -o fib_c fib_c.c
                  run:
                          ./fib_c.c 39
*/

File: rust/fib/fib.rs
// フィボナッチ数 by rust
use std::env;

fn fib(n: i64) -> i64 {
    if n < 2 {
            n
                } else {
                        fib(n - 1)  + fib(n - 2)
                            }
                            }

                            fn main() {
                                if let Some(s) = env::args().nth(1) {
                                        if let Ok(n) = s.parse::<i64>() {
                                                    println!("{}", fib(n))
                                                            }
                                                                }
                                                                }

                                                                /*
                                                                 example
                                                                     compile:
                                                                             rustc -C opt-level=3 fib_rust.rs
                                                                                 run:
                                                                                         ./fib_rust 39
                                                                                         */